metaflow/plugins/kubernetes/kubernetes_jobsets.py

```diff
@@ -0,0 +1,783 @@
+
+
+def _retrieve_replicated_job_statuses(jobset):
+    # We needed this abstraction because Jobsets changed thier schema
+    # in version v0.3.0 where `ReplicatedJobsStatus` became `replicatedJobsStatus`
+    # So to handle users having an older version of jobsets, we need to account
+    # for both the schemas.
+    if jobset.get("status", {}).get("replicatedJobsStatus", None):
+        return jobset.get("status").get("replicatedJobsStatus")
+    elif jobset.get("status", {}).get("ReplicatedJobsStatus", None):
+        return jobset.get("status").get("ReplicatedJobsStatus")
+    return None
+
+
+def _construct_jobset_logical_status(jobset, control_pod=None):
+    if not _basic_validation_for_js(jobset):
+        return JobsetStatus(
+            control_started=False,
+            control_completed=False,
+            workers_are_suspended=False,
+            workers_have_started=False,
+            all_jobs_are_suspended=False,
+            jobset_finished=False,
+            jobset_failed=False,
+            status_unknown=True,
+            jobset_was_terminated=False,
+            control_exit_code=None,
+            control_pod_status=None,
+            worker_pods_failed=False,
+            control_pod_failed=False,
+            some_jobs_are_running=False,
+        )
+
+    js_status = jobset.get("status")
+
+    control_started = False
+    control_completed = False
+    workers_are_suspended = False
+    workers_have_started = False
+    all_jobs_are_suspended = jobset.get("spec", {}).get("suspend", False)
+    jobset_finished = False
+    jobset_failed = False
+    status_unknown = False
+    jobset_was_terminated = False
+    worker_pods_failed = False
+    some_jobs_are_running = False
+
+    total_worker_jobs = [
+        w["replicas"]
+        for w in jobset.get("spec").get("replicatedJobs", [])
+        if w["name"] == "worker"
+    ][0]
+    total_control_jobs = [
+        w["replicas"]
+        for w in jobset.get("spec").get("replicatedJobs", [])
+        if w["name"] == "control"
+    ][0]
+
+    if total_worker_jobs == 0 and total_control_jobs == 0:
+        jobset_was_terminated = True
+
+    replicated_job_statuses = _retrieve_replicated_job_statuses(jobset)
+    for job_status in replicated_job_statuses:
+        if job_status["active"] > 0:
+            some_jobs_are_running = True
+
+        if job_status["name"] == "control":
+            control_started = job_status["active"] > 0 or job_status["succeeded"] > 0
+            control_completed = job_status["succeeded"] > 0
+            if job_status["failed"] > 0:
+                jobset_failed = True
+
+        if job_status["name"] == "worker":
+            workers_have_started = job_status["active"] == total_worker_jobs
+            if "suspended" in job_status:
+                # `replicatedJobStatus` didn't have `suspend` field
+                #  until v0.3.0. So we need to account for that.
+                workers_are_suspended = job_status["suspended"] > 0
```

Comment:
`suspend` was also not a supported keyword in Jobset [for version v0.2.0](https://github.com/kubernetes-sigs/jobset/blob/v0.2.0/api/jobset/v1alpha2/jobset_types.go#L92); This is why we need a conditional check over here.